Sources/Dollar.swift

```diff
@@ -561,6 +561,23 @@ public class $ {
         return (first / $.gcd(first, second)) * second
     }
     
+    /// Splits a collection into sets, grouped by the result of running each value through a callback.
+    ///
+    /// :param The array to group
+    /// :param callback Function whose response will be used as a key in the new string
+    public class func groupBy<T, U>(array: [T], callback: (T) -> U) -> [String: [T]]{
+        var grouped = [String: [T]]()
+        for element in array {
+            let key = String(callback(element))
+            if(grouped[key] != nil) {
+                grouped[key]!.append(element)
```

Comment:
Can you use the `if let` syntax instead of this.

```swift
if (let arr = grouped[key]) {
    arr.append(element)
} else {
    grouped[key] = [element]
}
```